From 25ea3d5abe53c07e035053c3ff590e765805b9d4 Mon Sep 17 00:00:00 2001
From: Joseph Hwang <jzh@fb.com>
Date: Tue, 17 May 2016 15:52:26 -0700
Subject: [PATCH] Added in Priority, General, and Replication queue sizes for
 RegionServers to the Web UI

Changed UI labels so that queue "size" refers to size in bytes and queue "length" refers to number of items in queue.
---
 .../hadoop/hbase/tmpl/regionserver/ServerMetricsTmpl.jamon   | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/regionserver/ServerMetricsTmpl.jamon b/hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/regionserver/ServerMetricsTmpl.jamon
index f5aa478..82cb4e7 100644
--- a/hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/regionserver/ServerMetricsTmpl.jamon
+++ b/hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/regionserver/ServerMetricsTmpl.jamon
@@ -193,14 +193,20 @@ MetricsHBaseServerWrapper mServerWrap;
 </%args>
 <table class="table table-striped">
 <tr>
-    <th>Compaction Queue Size</th>
-    <th>Flush Queue Size</th>
-    <th>Call Queue Size (bytes)</th>
+    <th>Compaction Queue Length</th>
+    <th>Flush Queue Length</th>
+    <th>Priority Call Queue Length</th>
+    <th>General Call Queue Length</th>
+    <th>Replication Call Queue Length</th>
+    <th>Total Call Queue Size (bytes)</th>
 
 </tr>
 <tr>
     <td><% mWrap.getCompactionQueueSize() %></td>
     <td><% mWrap.getFlushQueueSize() %></td>
+    <td><% mServerWrap.getPriorityQueueLength() %></td>
+    <td><% mServerWrap.getGeneralQueueLength() %></td>
+    <td><% mServerWrap.getReplicationQueueLength() %></td>
     <td><% TraditionalBinaryPrefix.long2String(mServerWrap.getTotalQueueSize(), "B", 1) %></td>
 </tr>
 </table>
-- 
2.8.0-rc2

